File: payroll-api/src/main/java/com/company/payroll/service/impl/StaffLeaveDetailsServiceImpl.java
package com.company.payroll.service.impl;

import com.company.payroll.mapper.StaffLeaveDetailsMapper;
import com.company.payroll.model.StaffLeaveDetails;
import com.company.payroll.service.StaffLeaveDetailsService;
import com.github.pagehelper.PageHelper;
import com.github.pagehelper.PageInfo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class StaffLeaveDetailsServiceImpl implements StaffLeaveDetailsService {

    @Autowired
    private StaffLeaveDetailsMapper staffLeaveDetailsMapper;

    @Override
    public PageInfo<StaffLeaveDetails> list(int page, int offset) {
        PageHelper.startPage(page, offset);
        return new PageInfo<StaffLeaveDetails>(staffLeaveDetailsMapper.selectList());
    }

    @Override
    public StaffLeaveDetails findById(Integer ldId) {
        return staffLeaveDetailsMapper.selectByPrimaryKey(ldId);
    }

    @Override
    public StaffLeaveDetails update(StaffLeaveDetails staffLeaveDetails) {
        staffLeaveDetailsMapper.updateByPrimaryKeySelective(staffLeaveDetails);
        return staffLeaveDetails;
    }
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/service/impl/StaffLeaveDetailsServiceImpl.kt
package com.company.payroll.service.impl

import com.company.payroll.mapper.StaffLeaveDetailsMapper
import com.company.payroll.model.StaffLeaveDetails
import com.company.payroll.service.StaffLeaveDetailsService
import com.github.pagehelper.PageHelper
import com.github.pagehelper.PageInfo
import org.springframework.beans.factory.annotation.Autowired

class StaffLeaveDetailsServiceImpl(@Autowired private val staffLeaveDetailsMapper: StaffLeaveDetailsMapper): StaffLeaveDetailsService {
  override fun list(page: Int, offset: Int): PageInfo<StaffLeaveDetails> {
    PageHelper.startPage<StaffLeaveDetails>(page, offset)
    return PageInfo<StaffLeaveDetails>(staffLeaveDetailsMapper.selectList())
  }

  override fun findById(ldId: Int): StaffLeaveDetails {
    return staffLeaveDetailsMapper.selectByPrimaryKey(ldId)
  }

  override fun update(staffLeaveDetails: StaffLeaveDetails): StaffLeaveDetails {
    staffLeaveDetailsMapper.updateByPrimaryKeySelective(staffLeaveDetails)
    return staffLeaveDetails
  }
}